### PR TITLE
Add Role base access to callback 

### PR DESCRIPTION
**Pull request by dharmendrak**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CallbackHandlerFactory.java [hunk 1]
```diff
@@ -11,20 +12,30 @@
...
     @Autowired(required = false)
-    public CallbackHandlerFactory(List<CallbackHandler> beans) {
+    public CallbackHandlerFactory(List<CallbackHandler> beans, UserRolesService userRolesService) {
+        this.userRolesService = userRolesService;
         beans.forEach(bean -> bean.register(eventHandlers));
     }
 
     public CallbackResponse dispatch(CallbackParams callbackParams) {
         return Optional.ofNullable(eventHandlers.get(callbackParams.getRequest().getEventId()))
+            .filter(h -> hasSupportedRoles(h, callbackParams.getParams().get(BEARER_TOKEN).toString()))
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CallbackHandlerFactory.java]: if the passed token doesn't have the authorised role, we should throw the relevant exception/Http status back.

Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CallbackHandlerFactory.java]: done as suggested

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CallbackHandlerFactory.java]: Thanks 👍

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CallbackHandlerFactory.java [hunk 2]
```diff
@@ -11,20 +12,35 @@
...
-                "Could not handle callback for event " + callbackParams.getRequest().getEventId()));
+            .orElseThrow(() -> new CallbackException("Could not handle callback for event " + eventId));
+    }
+
+    private boolean hasSupportedRoles(CallbackHandler callbackHandler, String authorisation, String eventId) {
+        List<String> userRoles = userService.getUserDetails(authorisation).getRoles();
+        if (callbackHandler.getSupportedRoles().stream().anyMatch(userRoles::contains)) {
+            return true;
+        } else {
+            throw new CallbackException("User does not have supported role for event " + eventId);
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CallbackHandlerFactory.java]: Should have been better to have specific Exception for this and mapped to HTTP Unauthorised response code. How about using `ForbiddenActionException` which is already mapped?

Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CallbackHandlerFactory.java]: done

**Pull request merged**